common/modules/token_send/src/token_send.rs

```diff
@@ -66,38 +86,36 @@ pub trait TokenSendModule {
         token: &TokenIdentifier,
         nonce: u64,
         amount: &BigUint,
-        opt_accept_funds_func: &OptionalArg<BoxedBytes>,
+        opt_accept_funds_func: &OptionalArg<ManagedBuffer>,
     ) -> SCResult<()> {
         if amount == &0u32 {
             return Ok(());
         }
 
-        let (function, gas_limit) = match opt_accept_funds_func {
-            OptionalArg::Some(accept_funds_func) => (
-                accept_funds_func.as_slice(),
-                self.transfer_exec_gas_limit().get(),
-            ),
-            OptionalArg::None => {
-                let no_func: &[u8] = &[];
-                (no_func, 0u64)
-            }
-        };
         let arg_buffer = ManagedArgBuffer::new_empty(self.type_manager());
-        let endpoint_name = ManagedBuffer::new_from_bytes(self.type_manager(), function);
         let mut payments = ManagedVec::new();
         payments.push(EsdtTokenPayment::new(token.clone(), nonce, amount.clone()));
 
+        let gas_limit: u64;
+        let function: ManagedBuffer;
+        let accept_funds_func = opt_accept_funds_func.clone().into_option();
+        if accept_funds_func.is_some() {
```

Comment:
I think it is cleaner to have this implementation
`
        let (function, gas_limit) = match opt_accept_funds_func {
            OptionalArg::Some(accept_funds_func) => (
                accept_funds_func.as_managed_ref(),
                self.transfer_exec_gas_limit().get(),
            ),
            OptionalArg::None => (ManagedBuffer::new().as_managed_ref(), 0u64),
        };
`

Reply:
that was my initial intention. couldn't figure out the syntax. you can commit this if you want. it is optional ofc

Reply:
+1